app/scripts/modules/core/src/pipeline/config/triggers/artifacts/artifact.component.ts

```diff
@@ -1,25 +1,59 @@
 import { IComponentController, IComponentOptions, module } from 'angular';
 
 import { PIPELINE_CONFIG_PROVIDER } from 'core/pipeline/config/pipelineConfigProvider';
-import { IExpectedArtifact, MissingArtifactPolicy } from 'core/domain/IExpectedArtifact';
+import { FieldType, IArtifactField, IExpectedArtifact, MissingArtifactPolicy } from 'core/domain/IExpectedArtifact';
 import { IPipeline } from 'core/domain/IPipeline';
 
+import { find } from 'lodash';
+
 class ArtifactController implements IComponentController {
   public artifact: IExpectedArtifact;
   public pipeline: IPipeline;
   public missingPolicies: string[] = Object.keys(MissingArtifactPolicy).map(key => MissingArtifactPolicy[key as any]);
+  public fieldNames: string[] = ['type', 'name', 'version', 'location', 'reference', 'artifactAccount', 'provenance', 'uuid'];
+  public fieldTypes: string[] = Object.keys(FieldType).map(key => FieldType[key as any]);
 
   public removeArtifact(): void {
     const artifactIndex = this.pipeline.expectedArtifacts.indexOf(this.artifact);
     this.pipeline.expectedArtifacts.splice(artifactIndex, 1);
 
     this.pipeline.triggers
-      .forEach(t => t.expectedArtifacts = t.expectedArtifacts.filter(a => !this.isEqual(a, this.artifact)));
+      .forEach(t => t.expectedArtifacts = t.expectedArtifacts.filter(a => !this.equals(a, this.artifact)));
+  }
+
+  public addField(): void {
+    const newField = {
```

Comment:
You should be able to just type this as `const newField: IArtifactField` instead of using `as IArtifactField` below.